src/models/time-sync.ts

```diff
@@ -224,64 +276,63 @@ export class TimeSyncModel implements ModelPluginInterface {
   async execute(inputs: ModelParams[]): Promise<ModelParams[]> {
     this.validateParams();
 
-    const dealer = await UnitsDealer();
-
     const pad = this.checkForPadding(inputs);
-    const paddedInputs = this.padInputs(inputs, pad, dealer);
-
-    return paddedInputs
-      .reduce((acc, input, index) => {
-        const currentMoment = moment(input.timestamp);
-
-        /** Checks if not the first input, then check consistency with previous ones. */
-        if (index > 0) {
-          const previousInput = paddedInputs[index - 1];
-          const previousInputTimestamp = moment(previousInput.timestamp);
-
-          /** Checks for timestamps overlap. */
-          if (
-            previousInputTimestamp
-              .add(previousInput.duration, 'seconds')
-              .isAfter(currentMoment)
-          ) {
-            throw new InputValidationError(INVALID_OBSERVATION_OVERLAP);
-          }
+    const paddedInputs = this.padInputs(inputs, pad);
+
+    const flattenInputs = paddedInputs.reduce((acc, input, index) => {
+      const currentMoment = moment(input.timestamp);
+
+      /** Checks if not the first input, then check consistency with previous ones. */
+      if (index > 0) {
+        const previousInput = paddedInputs[index - 1];
+        const previousInputTimestamp = moment(previousInput.timestamp);
+
+        /** Checks for timestamps overlap. */
+        if (
+          previousInputTimestamp
```

Comment:
`previousInputTimestamp.add()` updates the value of `previousTimestamp` in-place, breaking the cross-observation in-filling.

Reply:
replace with new `moment` instance instead to fix